catalyst_voices/lib/pages/registration/recover/seed_phrase/seed_phrase_input_panel.dart

```diff
@@ -78,7 +82,29 @@ class _SeedPhraseInputPanelState extends State<SeedPhraseInputPanel> {
   }
 
   Future<void> _uploadSeedPhrase() async {
-    // TODO(damian-molinski): Import implementation for KeychainCreation
+    final showUpload = await UploadSeedPhraseConfirmationDialog.show(context);
+    if (showUpload) {
+      await _showUploadDialog();
+    }
+  }
+
+  Future<void> _showUploadDialog() async {
+    final words = await UploadSeedPhraseDialog.show(context);
+
+    final isValid = SeedPhrase.isValid(
+      words: words,
+    );
+
+    if (isValid) {
+      _controller.words = words;
+    } else {
+      if (!mounted) return;
```

Comment:
Before using the controller we still need to check if the widget is mounted, otherwise the controller tries to do some logic which is only possible for non-disposed controller.

```suggestion
    if (!mounted) {
       return;
    } else if (isValid) {
      _controller.words = words;
    } else {
```